### PR TITLE
Add RSS for maps and tables

### PR DESCRIPTION
**Pull request by javierarce**
This PR adds RSS 2.0 and atom feeds for the public tables & maps.

### CODE REVIEW

**Comment by javierarce** [comment]
@rafatower: can you review this PR, please?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3294/))

**Comment by javisantana** [comment]
sorry, this needs to wait until we finish the refactor

**Comment by javierarce** [comment]
:(

**Comment by javisantana** [comment]
@javierarce will take care of that so move on

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3315/))

**Comment by saleiva** [comment]
:bump!

File: app/controllers/admin/pages_controller.rb [hunk 1]
```diff
@@ -171,6 +179,72 @@ def eligible_for_redirect?(user)
 
   private
 
+  def rss_feed(source = :maps, num_items = 25)
+    subdomain = CartoDB.extract_subdomain(request).strip.downcase
+
+    viewing_user = true
```
Comment (rafatower) [inline app/controllers/admin/pages_controller.rb]: we should extract this logic elsewhere

File: app/controllers/admin/pages_controller.rb [hunk 2]
```diff
@@ -171,6 +179,72 @@ def eligible_for_redirect?(user)
...
+
+    if source == :maps
+      @feed_items = organization.public_visualizations(1, num_items)
+    else
+      @feed_items = organization.public_datasets(1, num_items)
+    end
+
+    @feed_last_updated = nil
+    # So ugly, but current backend doesn't allows anything other than cycling with .each
+    @feed_items.each { |m|
```
Comment (rafatower) [inline app/controllers/admin/pages_controller.rb]: can't you use `@feed_last_updated = @feed_items.last.updated_at` ?

File: app/controllers/admin/pages_controller.rb [hunk 3]
```diff
@@ -171,6 +179,72 @@ def eligible_for_redirect?(user)
...
+                                                        type:     vis_type,
+                                                        per_page: num_items,
+                                                        privacy:  Visualization::Member::PRIVACY_PUBLIC,
+                                                        order:    'updated_at',
+                                                        o:        {updated_at: :desc},
+                                                        exclude_shared: true,
+                                                        exclude_raster: true,
+                                                      })
+
+    @feed_last_updated = nil
```
Comment (rafatower) [inline app/controllers/admin/pages_controller.rb]: same as above

File: spec/requests/admin/tables_spec.rb [hunk 4]
```diff
@@ -55,6 +55,20 @@ def app
     end
   end # GET /tables/:id
 
+  describe "GET feeds" do
+    it "returns an RSS feed" do
+      get "/datasets/feed", {}, :format => "rss"
+      last_response.status.should == 200
+      last_response.content_type.should eq("application/atom+xml; charset=utf-8")
```
Comment (rafatower) [inline spec/requests/admin/tables_spec.rb]: from my ignorance, shouldn't this be `application/rss+xml`?

Comment (rafatower) [inline spec/requests/admin/tables_spec.rb]: maybe this file is not being executed, check `Makefile`

**Comment by saleiva** [comment]
status on this?

**Comment by saleiva** [comment]
bump!

**Comment by saleiva** [comment]
bump!

**Comment by Kartones** [comment]
I'll check the status of the branch by merging latest master and deploying to staging.

@javisantana raised some privacy concerns if a user by mistake publishes/makes public a private map (as RSS get cached at RSS readers). For me is ok as we should strive/push for flooding every posible news/notifications system with our maps and not limit ourselves because of possible user mistakes, but please @saleiva confirm you're ok with that scenario and "known issue".

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4276/))

**Comment by saleiva** [comment]
uhm... veeery good concern I'd say. If the user makes the map private we will see a 404. The only problem is when the link is public with link I'd say. And for that you can always make it private or add a password. I'd say, lets go...

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4286/))

**Comment by javisantana** [comment]
it's ok

**Comment by Kartones** [comment]
ok, then releasing the real future of CartoDB, RSS!

**Comment by javierarce** [comment]
\o/ :moneybag: :moneybag: :moneybag:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4301/))

**Pull request merged**